src/compile/scale/type.ts

```diff
@@ -155,9 +158,13 @@ export function fieldDefMatchScaleType(specifiedType: ScaleType, fieldDef: Field
     return specifiedType === undefined || hasDiscreteDomain(specifiedType);
   } else if (type === Type.TEMPORAL) {
     if (!fieldDef.timeUnit) {
-      return contains([ScaleType.TIME, ScaleType.UTC, undefined], specifiedType);
-    } else {
       return contains([ScaleType.TIME, ScaleType.UTC, undefined], specifiedType) || hasDiscreteDomain(specifiedType);
+    } else {
+      if (isUTCTimeUnit(fieldDef.timeUnit)) {
+        return contains([ScaleType.UTC, undefined], specifiedType);
+      } else {
+        return contains([ScaleType.UTC, ScaleType.TIME, undefined], specifiedType);
```

Comment:
Why is UTC allow if timeUnit is not UTC? 

I guess that's because we want to change normalize timeUnit later?  If so, add comment.